x/incentive/keeper/bond.go

```diff
@@ -0,0 +1,101 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/umee-network/umee/v4/x/incentive"
+)
+
+// getAllBonded gets the total amount of a uToken bonded to an account across all three unbonding tiers
+func (k Keeper) getAllBonded(ctx sdk.Context, addr sdk.AccAddress, denom string) sdk.Coin {
+	bonded := sdk.NewCoin(denom, sdk.ZeroInt())
+	for _, tier := range []incentive.BondTier{incentive.BondTierLong, incentive.BondTierMiddle, incentive.BondTierShort} {
+		bonded = bonded.Add(k.GetBonded(ctx, addr, denom, tier))
+	}
+	return bonded
+}
+
+// accountBonds gets the total bonded and unbonding for a given account, as well as a list of ongoing unbondings,
+// for a single bond tier and uToken denom. Using incentive.BondTierUnspecified returns all tiers instead.
+// It reduces the returned bonded amount by completed unbondings without actually clearing those unbondings in state,
+// so it is safe for queries and any parts of Msg functions which are not intended to alter state.
+func (k Keeper) accountBonds(ctx sdk.Context, addr sdk.AccAddress, denom string, tier incentive.BondTier) (
+	bonded sdk.Coin, unbonding sdk.Coin, unbondings []incentive.Unbonding,
+) {
+	bonded = sdk.NewCoin(denom, sdk.ZeroInt())
```

Comment:
for zero coin can you use 
`coin.Zero(denom)` from util package, 
same for all zero valued coins

Reply:
fixed